backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inventory/InventoryRepository.java

```diff
@@ -465,6 +498,30 @@ public void saveInventoryLineHURecords(@NonNull final InventoryLine inventoryLin
 		}
 	}
 
+	public Optional<Quantity> getFreshBookedQtyFromStorage(@NonNull ProductId productId, @NonNull UomId inventoryLineUOMId, @NonNull HuId huId)
+	{
+		Optional<Quantity> bookedQty = Optional.empty();
+
+		I_M_HU hu = huDAO.getById(huId);
+
+		final I_M_HU_Storage huStorage = huStorageDAO.retrieveStorage(hu, productId);
```

Comment:
i think at this point we would be be better off using *not* the DB records, but hustorage, as sketched out in my last review